packages/solarwinds-apm/src/index.ts

```diff
@@ -48,6 +53,24 @@ if (!Reflect.has(globalThis, INIT)) {
       configurable: false,
       writable: false,
     })
+
+    let exited = false
+    Object.entries({
+      SIGINT: 2,
+      SIGTERM: 15,
+      beforeExit: -128,
+    }).map(([signal, code]) =>
+      process.once(signal, () => {
+        if (exited) return
+        exited = true
+
+        void Promise.all([TRACER_PROVIDER, METER_PROVIDER, LOGGER_PROVIDER])
+          .then((providers) =>
+            Promise.all(providers.map((provider) => provider?.shutdown())),
+          )
+          .finally(() => process.exit(128 + code))
```

Comment:
IIUC `beforeExit` is given -128 so the exit code becomes 0 since this is a normal termination scenario, correct?

Just quickly reading Node.js [reference](https://nodejs.org/api/process.html#signal-events) on signals there's the mention of default SIGINT and SIGTERM handlers being removed if a listener is set on the signal. Not sure what the "terminal mode" reset means, is it anything to be concerned with?

Reply:
The behaviour of the default handler is reproduced by the `process.exit` call. And afaik most terminal emulators will reset the mode themselves at exit so it's not really necessary. We also have to way to know what mode the terminal was initially in anyway.